Add support for Linux environments

tyro-apil opened the pull request:
# Testing
Tested the functionality in `Debian 12 Bookworm` and `Ubuntu 22.04 LTS`, and it worked. But, I haven't tested in other linux environments.

# Files for downloads
The URL for **scripts & binaries** are in the same form as in Windows scripts. But, these files `need to be added in specified endpoints` of the website.

tyro-apil committed:
feat: add linux compile target

tyro-apil committed:
Merge branch 'main' into linux

tyro-apil committed:
feat: add installer shell script

tyro-apil committed:
feat: add scheduler shell script

tyro-apil committed:
feat: add unscheduler shell script

tyro-apil committed:
feat: add README for setting up autologin in linux environment

bishal0602 commented:
@tyro-apil Thank you soo much for adding Linux support. I’ve added the necessary files to the specified endpoints on the server. Can you please test to confirm everything works as expected?

Really appreciate your effort on this!!

bishal0602 committed:
updated readme for linux support

tyro-apil commented:
@bishal0602  The script files in the website have `Windows line ending` i.e. `\r\n` which is not expected by LINUX as its line ending is `\n`. So, the script file couldn't be run.

bishal0602 committed:
Normalize line endings to LF for shell scripts

bishal0602 committed:
added .gitattributes to force unix line endings

bishal0602 commented:
@tyro-apil I’ve updated the script files to use Unix-style line endings (`\n`). Can you please test again to make sure it works correctly?

tyro-apil committed:
fix: remove sudoing installer script as it places the files in /root

tyro-apil committed:
chore: add some print lines before and after executing commands

tyro-apil commented:
@bishal0602 I've updated the install script in README to place the files in `user's HOME directory` instead of `/root`. Additionally, I have added some print statements in the installer script for displaying different states in the execution process of installer script.

bishal0602 commented:
@tyro-apil  I’ve updated the files in the endpoints with the latest changes from your updates. Let me know if there’s anything else to tweak.

tyro-apil committed:
fix: remove custom log file

tyro-apil committed:
fix: modify shebang for generalized path for bash

tyro-apil committed:
feat: add instruction to view logs of dispatcher

tyro-apil commented:
@bishal0602 I've made some final tweaks about logging.
